### PR TITLE
boards: particle_argon: Use GPIO_DT_SPEC for gpio handling

### PR DESCRIPTION
**Pull request by galak**
Convert gpio handling to use GPIO_DT_SPEC as this removes the use
of DT_LABEL.

Signed-off-by: Kumar Gala <galak@kernel.org>

### CODE REVIEW

**Comment by galak** [comment]
Will cleanup the other particle_* boards after we are good with this PR.

**Review by gmarull — COMMENTED** [review]
I think that refactor could go one step further by fixing some issues with this board init code, see my comments. It can be done in a separate PR though.

File: boards/arm/particle_argon/board.c [hunk 1]
```diff
@@ -11,28 +11,19 @@
 
 static inline void external_antenna(bool on)
```
[the remark this answers is not in the document]
Comment (galak) [reply boards/arm/particle_argon/board.c:12]: agreed, outside scope of this pr.

File: boards/arm/particle_argon/board.c [hunk 2]
```diff
@@ -11,28 +11,19 @@
...
-	gpio_pin_configure(pcb_gpio_dev, SKY_PCBn_GPIO_PIN,
-			   SKY_PCBn_GPIO_FLAGS
-			   | (on
-			      ? GPIO_OUTPUT_INACTIVE
-			      : GPIO_OUTPUT_ACTIVE));
+	gpio_pin_configure_dt(&ufl_gpio, (on ? GPIO_OUTPUT_ACTIVE : GPIO_OUTPUT_INACTIVE));
+	gpio_pin_configure_dt(&pcb_gpio, (on ? GPIO_OUTPUT_INACTIVE : GPIO_OUTPUT_ACTIVE));
 }
 
 static int board_particle_argon_init(const struct device *dev)
```
[the remark this answers is not in the document]
Comment (galak) [reply boards/arm/particle_argon/board.c:29]: agreed, outside scope of this pr.

**Review by gmarull — APPROVED** [review]

File: boards/arm/particle_argon/board.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (henrikbrixandersen) [reply boards/arm/particle_argon/board.c:14]: Great candidate for the regulator API, but I take it that's outside the scope of this PR :-)

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**